### DISCUSSION TITLE
The workflow is generated by dragging, rather than writing Python code.

### DISCUSSION THREAD

**Discussion by ChenpiDog** [apache/airflow#27284] [General]
### Description

I hope Airflow can provide drag and drop methods to establish workflows in the future, such as Kettle or DolphinScheduler, instead of writing python code.

### Use case/motivation

To improve the usability of airflow, although this change is great, it must be worthwhile. Airflow has done well, but I hope it can do better.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by tirkarthi** [comment]
This might require an AIP with less likely to be approved and airflow has core principle of workflow as code https://airflow.apache.org/docs/apache-airflow/stable/#workflows-as-code . You can try some project like windmill but it doesn't seem to support Airflow 2 https://github.com/mayansalama/windmill

> If you prefer clicking over coding, Airflow is probably not the right solution. The web interface aims to make managing workflows as easy as possible and the Airflow framework is continuously improved to make the developer experience as smooth as possible. However, the philosophy of Airflow is to define workflows as code so coding will always be required.

**Comment by potiuk** [comment]
yep it's not a feature. it's a dicussion. And huge one. And AIP.. In the meantime feel free to use DagFactory if you wnat to use Yaml.

I personally think Drag & Drop is a niche case and investment in making it will not pay. It might look like from your case, but when we see it as a product, the target users for drag & drop (i.e. mostly small teams without the need of versioning and sharing the code) is different than the current target users. So you might simply be mistaken about who the target user is. 

Something that seems cool for you does not necessary work when you have 10000 users. 

But all this should be discussed when someone takes on the task to justify and write a proposal.
